### PR TITLE
Avoid redirect to RDP when run is created from channels RHS 

### PR DESCRIPTION
**Pull request by trilopin**
## Summary
After a run is created through the modal, handle redir following:
- if user created the run in channels (home or run-list) and the run channel is the same where the user is -> open the new run in the RHS, refresh run list (don't leave channel)
- if user created the run in channels (home or run-list) and the run channel is NOT the same where the user is -> redirect to the related channel
- if the user created the run from backstage (pbe, pb list) -> redirect to RDP + refresh LHS

Added a bunch of e2e. 

## Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-playbooks/issues/1649

## Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

File: server/app/playbook_run_service.go [hunk 1]
```diff
@@ -218,10 +218,6 @@ func (s *PlaybookRunServiceImpl) sendWebhooksOnCreation(playbookRun PlaybookRun)
 
 // CreatePlaybookRun creates a new playbook run. userID is the user who initiated the CreatePlaybookRun.
 func (s *PlaybookRunServiceImpl) CreatePlaybookRun(playbookRun *PlaybookRun, pb *Playbook, userID string, public bool) (*PlaybookRun, error) {
-	if pb != nil && pb.ChannelMode == PlaybookRunLinkExistingChannel && pb.ChannelID == "" {
-		return nil, errors.New("channel id not specified to link the run")
-	}
-
```
Comment (lieut-data) [inline server/app/playbook_run_service.go:224]: I don't quite understand what this check used to do: can you elaborate?

File: webapp/src/browser_routing.ts [hunk 2]
```diff
@@ -23,6 +24,15 @@ export const navigateToPluginUrl = (urlPath: string) => {
     WebappUtils.browserHistory.push(pluginUrl(urlPath));
 };
 
+/**
+ * Navigate to channel given a channelId and teamName
+ * Note that channel will be retrieved from the server to get the name.
+ */
+export const navigateToChannel = async (teamName: string, channelId: string) => {
+    const channel = await Client4.getChannel(channelId);
```
[the remark this answers is not in the document]
Comment (lieut-data) [reply webapp/src/browser_routing.ts:32]: I think we actually support just passing the `id`, e.g. `/test/channels/4udb87gnitbffrzbuau4q3u34w` resolves to `/test/channels/asdasd` for my localhost.

**Review by lieut-data — APPROVED** [review]
This is super slick! Love the intelligent approach to redirecting -- a handful of comments, but nothing strictly blocking :)

**Review by gbochora — APPROVED** [review]
Great work on tests! Heads up, I've pushed a commit to fix comments in e2e tests.

**Comment by trilopin** [comment]
> Great work on tests! Heads up, I've pushed a commit to fix comments in e2e tests.

Thanks, @gbochora I'm crazy bad at writing that comments 🤦🏽

**Pull request merged**